docs/reference/scripting/using.asciidoc

```diff
@@ -1,425 +1,419 @@
---------------------------------------------------
-GET my-index-000001/_search
+----
+POST /_scripts/painless/_execute
 {
-  "query": {
-    "script_score": {
-      "query": {
-        "match": {
-            "message": "some message"
-        }
-      },
-      "script": {
-        "id": "calculate-score",
-        "params": {
-          "my_modifier": 2
-        }
-      }
+  "script": {
+    "source": "params['hello_world']",
+    "params": {
+        "hello_world": "Hello, world!"
     }
   }
 }
---------------------------------------------------
-// TEST[continued]
+----
 
-And deleted with:
+This version of the script removes several components:
 
-[source,console]
------------------------------------
-DELETE _scripts/calculate-score
------------------------------------
-// TEST[continued]
+* The `return` keyword. Painless automatically uses the final statement in a
+script (when possible) to produce a return value in a script context that
+requires one.
+* The semicolon at the end of the `source` statement. Painless does not
+require semicolons for the final statement of a block. However, it does require
+them in other cases to remove ambiguity.
+* The `get` method, which is replaced with brackets `[]`. Painless
+uses a shortcut specifically for the `Map` type that allows us to use brackets
+instead of the lengthier `get` method.
 
 [discrete]
-[[modules-scripting-search-templates]]
-=== Search templates
-You can also use the `_scripts` API to store **search templates**. Search
-templates save specific <<search-search,search requests>> with placeholder
-values, called template parameters.
+[[script-whats-next]]
+=== What's next
+Now that you've learned the basic syntax of a Painless script, we'll dig more
+into the language itself. Understanding the building blocks of the language
+will help you to construct more complex and powerful scripts.
 
-You can use stored search templates to run searches without writing out the
-entire query. Just provide the stored template's ID and the template parameters.
-This is useful when you want to run a commonly used query quickly and without
-mistakes.
+<<script-spec-types,Learn about types>>
 
-Search templates use the https://mustache.github.io/mustache.5.html[mustache
-templating language]. See <<search-template>> for more information and examples.
+[[script-spec-types]]
+=== Types
+A type is a classification of data used to define the properties of a value.
+These properties specify what data a value represents and the rules for how a
+value is evaluated during an operation. Each type belongs to one of the
+following categories: primitive, reference, or dynamic.
 
 [discrete]
-[[modules-scripting-using-caching]]
-=== Script caching
+[[script-type-primitive]]
+==== Primitive types
+A primitive type represents basic data built natively into the JVM and is
+allocated to non-heap memory. Examples of
+<<available-primitive-types,available primitive types>> are `byte`, `int`, and
+`long`.
```

Comment:
I wonder if this is too much info/advanced for a beginner's guide. This kind of info might better be left to the spec.

Reply:
@jdconrad, agree 💯 I'll start paring back the details to keep this introduction lighter.